content/articles/build-and-dockerize-a-full-stack-react-app-with-nodejs-and-nginx/index.md

```diff
@@ -0,0 +1,673 @@
+
+### Prerequisites
+
+- Have [Node.js](https://nodejs.org/en/download/) and [Docker](https://www.docker.com/products/docker-desktop) installed on your computer.
+- Prior knowledge on how to use Docker and Docker-compose to run and manage Docker containers
+- Good understanding of using [React](https://www.youtube.com/watch?v=w7ejDZ8SWv8) to create front-end templates
+- Understand How to [create API services using the Node.js](https://www.youtube.com/watch?v=Oe421EPjeBE) framework.
+
+### Setup project basics
+
+This involves two major setups a client and back-end API services. The application involves multi-stage building. You utilize several FROM statements in your Dockerfile for multi-stage builds. Then write a docker-compose.yml file to execute all the Dockerfiles with the proper instructions to set a fully dockerized application. To start with, create a project directory to work on. We will start by creating a Node.js back-end API with Express to expose a REST API and communicate with the MySQL database.
+
+### Build the back-end Node.js application
+
+Node.js will help us to process all tractions and communication that we need to make to the database. In this case, we will use a CRUD use case to implement a simple Node.js Rest API.
+
+#### Step1: Project structure
+
+Inside your project directory, go ahead and create a new folder and name it a server. Will write our backed login inside this directory. So once created cd to this folder and initialize a Node.js application with `npm init -y`. Once executed, a `package.json` file will be created to add any dependencies that we need our application to use.
+
+#### Step2: Project dependencies
+
+Here we will use the following libraries.
+
+- CORS - a technique that permits restricted resources on a web page to be accessed from a domain different than the one from which the initial resource was provided, a case that occurs when you are creating an Express API.
+- Express - a Node.js framework that enables us to create API endpoints to request data from the server, a process that requests and sends a response with the requested information. Express will help us write different functions that will help us process different CRUD requests.
+- Mysql2 - will enable us to write different functions that will help us access MySQL Server and execute SQL queries from different requests.
+
+Run the following command (inside the server folder) to install the above packages
+
+```bash
+npm install cors express mysql2
+```
+
+#### Step3: Writing the API configurations
```

Comment:
You can probably give idea on what is the objective of this application. What are you trying to build?